app/api/v2/management/service_accounts.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module API::V2
+  module Management
+    # ServiceAccounts server-to-server API
+    class ServiceAccounts < Grape::API
+      helpers ::API::V2::NamedParams
+
+      desc 'ServiceAccounts related routes'
+      resource :service_accounts do
+
+        desc 'Get specific service_account information' do
+          @settings[:scope] = :read_service_accounts
+          success API::V2::Entities::ServiceAccounts
+        end
+        params do
+          optional :uid, type: String, allow_blank: false, desc: 'service_account uid'
+          optional :email, type: String, allow_blank: false, desc: 'service_account email'
+          exactly_one_of :uid, :email
+        end
+        post '/get' do
+          declared_params = declared(params, include_missing: false)
+
+          service_acc = ServiceAccount.find_by!(declared_params)
+          error!('Service account doesnt exist', 422) unless service_acc
+
+          present service_acc, with: API::V2::Entities::ServiceAccounts
+          status 200
+        end
+
+        desc 'Get service_accounts as a paginated collection' do
+          @settings[:scope] = :read_service_accounts
+          success API::V2::Entities::ServiceAccounts
+        end
+        params do
+          use :pagination_filters
+        end
+        post '/list' do
+          service_accs = ServiceAccount.all
+          service_accs.tap { |q| present paginate(q), with: API::V2::Entities::ServiceAccounts }
+          status 200
+        end
+
+        desc 'Create service_account' do
+          @settings[:scope] = :write_service_accounts
+          success API::V2::Entities::ServiceAccounts
+        end
+        params do
+          requires :service_account_uid, type: String, allow_blank: false, desc: 'service_account uid'
```

Comment:
As soon, as there is no user side ability to create a service account, we allow full customization for the server logic.
Maybe its a good idea to add both ways, allow to pass UID, but if its nil - we can generate internally.
Works for you?

Reply:
I got your point. Great! 
I like this idea)

I think for internal generating we can define some prefix, that will differ with User one( E.g. "SA" instead of "ID")

I'm scared that, maybe user only signed up, without login. In this case, we don't have such User on Applogic. And we can pass you an uid, that's already exists on Barong

Reply:
@dpatsora yeah, dont be so scared, my little. There is a check inside ServiceAccount model. We always check if UID and EMAIL are uniq in both tables  - Users and ServiceAccounts

Reply:
I'm scared not about Barong)
I just don't want to send one more request from Applogic and do it in loop)